### PR TITLE
tewisay: init at git-2017-04-14

### PR DESCRIPTION
**Pull request by ghost**
###### Motivation for this change
I like cowsay, so I wanted to add a fun [alternative](https://github.com/lucy/tewisay) to this repository.
I'm very new to this, so if I'm not following protocol correctly, please just tell me what to do.
Making a nixpkg is a tad confusing to a beginner.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/tools/misc/tewisay/default.nix [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{ stdenv, buildGoPackage, fetchgit, asciidoc, go-runewidth, pflag }:
```
Comment (nlewo) [inline pkgs/tools/misc/tewisay/default.nix]: Your package fails to build.
You have to specify Go dependencies by using the `goDeps` attributes. See   https://nixos.org/nixpkgs/manual/#ex-buildGoPackage-3
These dependencies can be generated by some tools such as go2nix.

File: pkgs/development/tools/go-runewidth/default.nix [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    url = https://github.com/mattn/go-runewidth.git;
+    rev = "97311d9f7767e3d6f422ea06661bc2c7a19e8a5d";
+    sha256 = "0dxlrzn570xl7gb11hjy1v4p3gw3r41yvqhrffgw95ha3q9p50cg";
+  };
+
+  meta = {
+    description = "Provides functions to get fixed width of the character or string";
+    homepage = https://github.com/mattn/go-runewidth;
+    platforms = stdenv.lib.platforms.all;
+    maintainers = [ stdenv.lib.maintainers.rob ];
```
Comment (nlewo) [inline pkgs/development/tools/go-runewidth/default.nix]: Are you `rob` ? :/

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -2412,6 +2412,8 @@ with pkgs;
 
   go-mtpfs = callPackage ../tools/filesystems/go-mtpfs { };
 
+  go-runewidth = callPackage ../development/tools/go-runewidth { };
```
Comment (nlewo) [inline pkgs/top-level/all-packages.nix]: Useless since this is a Go dep

File: pkgs/top-level/all-packages.nix [hunk 4]
```diff
@@ -4024,6 +4026,8 @@ with pkgs;
 
   pfstools = callPackage ../tools/graphics/pfstools { };
 
+  pflag = callPackage ../development/tools/pflag { };
```
Comment (nlewo) [inline pkgs/top-level/all-packages.nix]: Useless since this is a go dependency.

**Review by nlewo — CHANGES_REQUESTED** [review]
Welcome to the Nix packaging! 
I will try to help you in this process.

1. Squash your commits (no merge commit in your PR)
2. The format of a package which comes from a commit is `package-name-YYYY-MM-DD`. See https://nixos.org/nixpkgs/manual/#sec-package-naming
3. Set the licence of the package you are adding.

File: pkgs/development/tools/go-runewidth/default.nix (hunk 2 above)
Comment (Chiiruno) [reply pkgs/development/tools/go-runewidth/default.nix]: I wasn't sure if I could change that. Do I have to add my name to stdenv.lib.maintainers or can I just add it there?

**Comment by Chiiruno** [comment]
I'm going to close this and then reopen with a new fork, I'm having a hard time with squashing and rebasing.

**Pull request closed without merging**